Objects/memoryobject.c

```diff
@@ -956,15 +965,17 @@ PyMemoryView_GetContiguous(PyObject *obj, int buffertype, char order)
 memoryview.__new__
 
     object: object
+    *
+    _flags: int(c_default='PyBUF_FULL_RO') = PyBUF_FULL_RO
 
 Create a new memoryview object which references the given object.
 [clinic start generated code]*/
 
 static PyObject *
-memoryview_impl(PyTypeObject *type, PyObject *object)
-/*[clinic end generated code: output=7de78e184ed66db8 input=f04429eb0bdf8c6e]*/
+memoryview_impl(PyTypeObject *type, PyObject *object, int _flags)
```

Comment:
Did you consider adding a new function in `_testcapi` which is same as memoryview and accepts flags to avoid altering the builtin? This change will probably make `memoryview` signature look worse in `help` or inspection.

Reply:
I couldn't get that to work because the function it needs to call isn't part of the public C API, so the alternative was to add a new public C API function, which felt like it might be worse.

Also, thanks for your review! I'll address the other comments later today.

Reply:
Perhaps move it to a classmethod, so it’s more hidden?

Reply:
Yes that's a bit cleaner, done.